unittest/fortran/CMakeLists.txt

```diff
@@ -0,0 +1,30 @@
+include(CheckGeneratorSupport)
+if(NOT CMAKE_GENERATOR_SUPPORT_FORTRAN)
+  message(STATUS "Skipping Tests for the LAMMPS Fortran Module: no Fortran support in build tool")
+  return()
+endif()
+
+include(CheckLanguage)
+check_language(Fortran)
+if(CMAKE_Fortran_COMPILER)
+  enable_language(Fortran)
+  get_filename_component(LAMMPS_FORTRAN_MODULE ${LAMMPS_SOURCE_DIR}/../fortran/lammps.f90 ABSOLUTE)
+  if(BUILD_MPI)
+    find_package(MPI REQUIRED)
+  else()
+    add_library(fmpi_stubs STATIC mpi_stubs.f90)
+    add_library(MPI::MPI_Fortran ALIAS fmpi_stubs)
+  endif()
+
+  add_library(flammps STATIC ${LAMMPS_FORTRAN_MODULE})
```

Comment:
I think it would be good to move `flammps` to the main cmake and hide it behind an option like `ENABLE_FORTRAN_INTERFACE`.

Reply:
Why? That library is a "reverse interface" only required to do testing of Fortran code from the C++ googletest library. Also there is no "enabling" of the Fortran interface. The module is distributed as source code and needs to be compiled along with the target application using the desired Fortran compiler, so that flag would be *very* misleading.

Reply:
You are right, we don't really need an option, if there is Fortran compiler after ` check_language(Fortran)` just compile it. I think the `libflammps` could be useful for applications in the same as `liblammps` is. And for gfortran the `mod` files are versioned.

Reply:
No. Please don't make these statements unless you have read the docs about the Fortran interface and how to use it. The library compiled here is for testing use only and meant to be that way.

You are not getting my point. The versioning in gfortran .mod files one of the reasons why Fortran modules are a major PITA for binary distribution. The by far cleanest way to deal with is IMNSHO is to distribute the module as source and engineer it so that this is a single fully self-contained file that wraps the C library. Luckily, this seems to be possible in this case.

Reply:
For ensuring version compatibility, we use [shacl](https://github.com/shacl/cmake) for Fortran libraries at LANL, but as quick replacement one can just check if the compiler ID and version is the same when using the exported CMake target (in `LAMMPSConfig.cmake.in`).

In general, it seems a big unnatural to me to ask the user to add a source from another code to his own. Hence I was suggesting make it a normal (Fortran) library. Another option would be to make `flammps` an `INTERFACE` library with `lammps.f90` being a public source. In both cases CMake would take care of everything for the user. But this is out of the scope for this PR, so let's discuss this in another PR once this is merged.

Reply:
There is nothing to discuss. My mind is set up and I refuse to spend more time on this.